WordPress/src/main/java/org/wordpress/android/ui/media/MediaPickerActivity.java

```diff
@@ -0,0 +1,455 @@
+
+    /**
+     * Pass a {@link String} with this key in the {@link android.content.Intent} to set the title.
+     */
+    public static final String ACTIVITY_TITLE_KEY           = "activity-title";
+    /**
+     * Pass an {@link java.util.ArrayList} of {@link org.wordpress.mediapicker.source.MediaSource}'s
+     * in the {@link android.content.Intent} to set image sources for selection.
+     */
+    public static final String IMAGE_MEDIA_SOURCES_KEY      = "image-media-sources";
+    /**
+     * Pass an {@link java.util.ArrayList} of {@link org.wordpress.mediapicker.source.MediaSource}'s
+     * in the {@link android.content.Intent} to set video sources for selection.
+     */
+    public static final String VIDEO_MEDIA_SOURCES_KEY      = "video=media-sources";
+    /**
+     * Key to extract the {@link java.util.ArrayList} of {@link org.wordpress.mediapicker.MediaItem}'s
+     * that were selected by the user.
+     */
+    public static final String SELECTED_CONTENT_RESULTS_KEY = "selected-content";
+
+    private static final long   TAB_ANIMATION_DURATION_MS = 250l;
+    private static final String TAB_TITLE_IMAGES          = "Images";
+    private static final String TAB_TITLE_VIDEOS          = "Videos";
+
+    private MediaPickerAdapter     mMediaPickerAdapter;
+    private ArrayList<MediaSource> mImageSources;
+    private ArrayList<MediaSource> mVideoSources;
+    private SlidingTabLayout       mTabLayout;
+    private WPViewPager            mViewPager;
+    private ActionMode             mActionMode;
+    private String                 mCapturePath;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        lockRotation();
+        addMediaSources();
+        setTitle();
+        initializeContentView();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.media_picker, menu);
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.capture_image) {
+            MediaUtils.launchCamera(this, new MediaUtils.LaunchCameraCallback() {
+                @Override
+                public void onMediaCapturePathReady(String mediaCapturePath) {
+                    mCapturePath = mediaCapturePath;
+                }
+            });
+            return true;
+        } else if (item.getItemId() == R.id.capture_video) {
+            MediaUtils.launchVideoCamera(this);
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        switch (requestCode) {
+            case MediaUtils.RequestCode.ACTIVITY_REQUEST_CODE_TAKE_PHOTO:
```

Comment:
I noticed that after I snapped an image and added it to a post, it didn't show in the media picker when I returned to it. IIRC you need to pass an intent to the android gallery in order for it to show there.